ts/DataGrid/DataGrid.ts

```diff
@@ -1191,16 +1185,44 @@ class DataGrid {
         height?: number | string | null
     ): void {
         if (width) {
-            this.innerContainer.style.width = width + 'px';
+            if (typeof width === 'string') {
+                this.innerContainer.style.width = width;
+            } else {
+                this.innerContainer.style.width = width + 'px';
+            }
         }
 
         if (height) {
-            this.gridContainer.style.height = height + 'px';
-            this.outerContainer.style.height = height + 'px';
+            if (typeof height === 'string') {
+                this.outerContainer.style.height = height;
+            } else {
+                this.outerContainer.style.height =
+                    `${height} - ${this.getHeaderHeight() + this.getMarginHeight(height)}px`;
+            }
         }
 
         this.render();
     }
+
+    /**
+     * If the grid is in the parent container that has margins, calculate the
+     * height of the margins.
+     * @internal
+     *
+     * @param  height
+     * The height of the parent container.
+     */
+    private getMarginHeight(height: number): number {
+        return height - this.gridContainer.getBoundingClientRect().height;
```

Comment:
How about using `getStyle()` with the margin or padding param?

Reply:
I can't get the correct values using the `getStyle` method.  I'm afraid it has to stay as it is.